app/models/task.py

```diff
@@ -1,6 +1,68 @@
 from flask import current_app
 from app import db
-
+from datetime import datetime
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)  
+    title = db.Column(db.String) # name of task or title of task
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    # adding one to many relationship  tasks to goal ==> dog to person 
+    # nullable = task might not belong to a goal
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable=True) 
+
+    def to_json_response(self):
+        '''
+        Converts a Task instance into JSON
+        Output: Returns a Python dictionary in the shape of JSON response 
+        that the API returns in the route that is called (GET).
+        '''
+        return {"task": 
+                        {"id": self.id,
+                        "title": self.title,
+                        "description": self.description,
+                        "is_complete": bool(self.completed_at)}
+                }
+                
+    def task_to_json_response(self):
+        '''
+        Converts a Task's instance into JSON response
+        Output: Returns a Python dictionary in the shape JSON response
+        for only one task.
+        '''
+        return {"id": self.id,
+                "title": self.title,
+                "description": self.description,
+                "is_complete": bool(self.completed_at)}
+
+    def task_to_json_response_w_goal(self):
```

Comment:
Great work encapsulating logic in methods in the model. In this one, you make use of `task_to_json_response()` as I suggested above. Nice work!